tools/souper-check.cpp

```diff
@@ -134,87 +137,96 @@ int SolveInst(const MemoryBufferRef &MB, Solver *S) {
-        llvm::outs() << "known negative from souper: "
-                     << convertToStr(Negative) << "\n";
-        ++Success;
+    if (isInferDFA()) {
+      if (InferNeg) {
+        bool Negative;
+        if (std::error_code EC = S->negative(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                             Negative, IC)) {
+          llvm::errs() << "Error: " << EC.message() << '\n';
+          Ret = 1;
+          ++Error;
+        } else {
+          llvm::outs() << "known negative from souper: "
+                       << convertToStr(Negative) << "\n";
+          ++Success;
+        }
       }
-    } else if (InferNonNeg) {
-      bool NonNegative;
-      if (std::error_code EC = S->nonNegative(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
-                                              NonNegative, IC)) {
-        llvm::errs() << "Error: " << EC.message() << '\n';
-        Ret = 1;
-        ++Error;
-      } else {
-        llvm::outs() << "known nonNegative from souper: "
-                     << convertToStr(NonNegative) << "\n";
-        ++Success;
+      if (InferNonNeg) {
+        bool NonNegative;
+        if (std::error_code EC = S->nonNegative(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                                NonNegative, IC)) {
+          llvm::errs() << "Error: " << EC.message() << '\n';
+          Ret = 1;
+          ++Error;
+        } else {
+          llvm::outs() << "known nonNegative from souper: "
+                       << convertToStr(NonNegative) << "\n";
+          ++Success;
+        }
       }
-    } else if (InferKnownBits) {
-      unsigned W = Rep.Mapping.LHS->Width;
-      KnownBits Known(W);
-      if (std::error_code EC = S->knownBits(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
-                                            Known, IC)) {
-        llvm::errs() << "Error: " << EC.message() << '\n';
-        Ret = 1;
-        ++Error;
-      } else {
-        llvm::outs() << "knownBits from souper: "
-                     << Inst::getKnownBitsString(Known.Zero, Known.One) << "\n";
-        ++Success;
+      if (InferKnownBits) {
+        unsigned W = Rep.Mapping.LHS->Width;
+        KnownBits Known(W);
+        if (std::error_code EC = S->knownBits(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
+                                              Known, IC)) {
+          llvm::errs() << "Error: " << EC.message() << '\n';
+          Ret = 1;
+          ++Error;
+        } else {
+          llvm::outs() << "knownBits from souper: "
+                       << Inst::getKnownBitsString(Known.Zero, Known.One) << "\n";
+          ++Success;
+        }
       }
-    } else if (InferPowerTwo) {
-      bool PowTwo;
-      if (std::error_code EC = S->powerTwo(Rep.BPCs, Rep.PCs, Rep.Mapping.LHS,
-                                           PowTwo, IC)) {
-        llvm::errs() << "Error: " << EC.message() << '\n';
-        Ret = 1;
-        ++Error;
-      } else {
+      if (InferPowerTwo) {
+        bool PowTwo;
```

Comment:
do you see the darker green here? I think github is telling us about a whitespace problem, could you please check this?

Reply:
I don't see a problem here, it may be a diff glitch.

Reply:
It's actually my fault. Got it fixed.